counsel.el

```diff
@@ -6502,16 +6548,46 @@ specified by the `blddir' property."
              (compile cmd)
           (remove-hook 'compilation-start-hook #'counsel-compile--update-history))))))
 
+(defun counsel-compile-edit-command ()
+  "Insert current compile command into mini-buffer for editing.
+
+This mirrors the behaviour of `ivy-insert-current' but with specific
+handling for the counsel-compile metadata."
```

Comment:
Nits:
* "Minibuffer" isn't usually hyphenated.
* Ivy follows the Emacs convention of US spelling, so behaviour -> behavior.
* Quote `counsel-compile` with backtick/apostrophe.